src/workerd/api/pyodide/eval.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+
+namespace workerd::api::pyodide {
+
+class DynEvalImpl final: public jsg::Object {
+public:
+  void enableEval(jsg::Lock& js) {
+    js.setAllowEval(true);
```

Comment:
I think this is not the best place for it since it has to be imported by an internal builtin JS module. So any internal module that imports it should be responsible for making sure they don't leak it improperly. I added an extra callee check in `builtin_wrappers.js` [see here](https://github.com/cloudflare/workerd/pull/1467/files#r1414626194).

Reply:
Rather than this, I'd prefer an approach where this exports a function like...

```
impl.eval(() => { /** ... **/ });
```

Where the `js.setAllowEval(false)` is automatically handled when the call to `impl.eval(...)` completes.